src/ncp/ncp_uart.cpp

```diff
@@ -256,7 +264,15 @@ void NcpUart::HandleFrame(void *aContext, uint8_t *aBuf, uint16_t aBufLength)
 
 void NcpUart::HandleFrame(uint8_t *aBuf, uint16_t aBufLength)
 {
+#if OPENTHREAD_ENABLE_NCP_SPINEL_ENCRYPTER
+    size_t bufTransformedLen = sizeof(mRxBuffer);
+    if (SpinelEncrypter::DecryptInbound(aBuf, aBufLength, aBuf, &bufTransformedLen))
```

Comment:
We are passing same `aBuf` for input/output to `Decrypt/Encrypt` methods which would then do in-place operation.
How about if we change the definitions to expect a single buffer and also pass it the max size of the buffer as argument

```
// aFrameBuf : Pointer to buffer containing the frame, also where the encrypted frame will be placed
// aFrameSize : Max number of bytes in frame buffer.
// aFrameLength: Pointer to store frame length, on input value is set to frame length, 
//              on output changed to show the frame length after decryption.
bool EncryptOutbound(unsigned char *aFrameBuf, size_t aFrameSize,  size_t *aFrameLength)
````

Thoughts?

Reply:
I agree, will do as you described.

Reply:
Thanks.

Reply:
Done.